samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/ChatParticipantController.cs

```diff
@@ -0,0 +1,114 @@
+        ChatParticipantRepository chatParticipantRepository,
+        ChatSessionRepository chatSessionRepository)
+    {
+        this._logger = logger;
+        this._chatParticipantRepository = chatParticipantRepository;
+        this._chatSessionRepository = chatSessionRepository;
+    }
+
+    /// <summary>
+    /// Join a use to a chat session given a chat id and a user id.
+    /// </summary>
+    /// <param name="messageRelayHubContext">Message Hub that performs the real time relay service.</param>
+    /// <param name="chatParticipantParam">Contains the user id and chat id.</param>
+    [HttpPost]
+    [Route("chatParticipant/join")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> JoinChatAsync(
+        [FromServices] IHubContext<MessageRelayHub> messageRelayHubContext,
+        [FromBody] ChatParticipant chatParticipantParam)
+    {
+        string userId = chatParticipantParam.UserId;
+        string chatId = chatParticipantParam.ChatId;
+
+        // Make sure the chat session exists.
+        try
+        {
+            await this._chatSessionRepository.FindByIdAsync(chatId);
+        }
+        catch (Exception ex) when (ex is KeyNotFoundException || ex is ArgumentOutOfRangeException)
+        {
+            return this.BadRequest("Chat session does not exist.");
+        }
+
+        // Make sure the user is not already in the chat session.
+        if (await this._chatParticipantRepository.IsUserInChatAsync(userId, chatId))
+        {
+            return this.BadRequest("User is already in the chat session.");
+        }
+
+        var chatParticipant = new ChatParticipant(userId, chatId);
+        await this._chatParticipantRepository.CreateAsync(chatParticipant);
+
+        // Broadcast the user joined event to all the connected clients.
+        // Note that the client who initiated the request may not have joined the group.
+        await messageRelayHubContext.Clients.Group(chatId).SendAsync(UserJoinedClientCall, chatId, userId);
+
+        return this.Ok(chatParticipant);
+    }
+
+    /// <summary>
+    /// Get a list of chat participants that have the same chat id.
+    /// </summary>
+    /// <param name="chatId">The Id of the chat to get all the participants from.</param>
+    [HttpGet]
+    [Route("chatParticipant/getAllParticipants/{chatId:guid}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
+    public async Task<IActionResult> GetAllParticipantsAsync(Guid chatId)
+    {
+        // Make sure the chat session exists.
+        try
+        {
+            await this._chatSessionRepository.FindByIdAsync(chatId.ToString());
+        }
+        catch (Exception ex) when (ex is KeyNotFoundException || ex is ArgumentOutOfRangeException)
+        {
+            return this.BadRequest("Chat session does not exist.");
```

Comment:
We have a separate task to standardize the route pattern. We can come up with the appropriate route in that other task all together.

Reply:
in this case the ID is valid (since it passes the GUID validation) but it doesn't map to a session. seems like the same case as `GetChatMessagesAsync` and `GetSourcesAsync` in `ChatHistoryController`